app/src/main/java/org/p2p/wallet/send/ui/search/NewSearchFragment.kt

```diff
@@ -0,0 +1,177 @@
+
+                        override fun onQueryTextChange(newText: String?): Boolean {
+                            onSearchQueryChanged(newText.orEmpty())
+                            return true
+                        }
+                    },
+                    menuRes = R.menu.menu_search_with_scan,
+                    searchHintRes = R.string.search_edittext_hint
+                )
+                searchView?.setOnFocusChangeListener { v, hasFocus ->
+                    if (!hasFocus) toggleSearchView()
+                }
+                setNavigationOnClickListener { popBackStack() }
+                setOnMenuItemClickListener {
+                    if (it.itemId == R.id.itemScan) {
+                        presenter.onScanClicked()
+                        true
+                    } else {
+                        false
+                    }
+                }
+            }
+
+            buttonScan.setOnClickListener { presenter.onScanClicked() }
+            buttonContinue.setOnClickListener {
+                presenter.onContinueClicked(
+                    toolbar.searchView?.query?.toString().orEmpty()
+                )
+            }
+
+            buttonPaste.setOnClickListener { pasteText() }
+
+            recyclerView.layoutManager = LinearLayoutManager(requireContext())
+            recyclerView.attachAdapter(searchAdapter)
+        }
+
+        presenter.loadInitialData()
+    }
+
+    private fun onSearchQueryChanged(newQuery: String) {
+        presenter.search(newQuery)
+    }
+
+    override fun showLoading(isLoading: Boolean) {
+        binding.progressBar.isInvisible = !isLoading
+    }
+
+    override fun showNotFound() = with(binding) {
+        textViewNotFoundTitle.isVisible = true
+        groupErrorView.isVisible = false
+        groupEmptyView.isVisible = false
+        recyclerView.isVisible = false
+    }
+
+    override fun showEmptyState(isEmpty: Boolean) = with(binding) {
+        textViewNotFoundTitle.isVisible = false
+        groupErrorView.isVisible = false
+        groupEmptyView.isVisible = isEmpty
+        recyclerView.isVisible = !isEmpty
+    }
+
+    override fun showErrorState(isButtonEnabled: Boolean) = with(binding) {
+        groupErrorView.isVisible = true
+        buttonContinue.isEnabled = isButtonEnabled
+        groupEmptyView.isVisible = false
+        recyclerView.isVisible = false
+        textViewNotFoundTitle.isVisible = false
+    }
+
+    override fun showSearchValue(value: String) {
+        binding.toolbar.searchView?.setQuery(value, true)
+    }
+
+    override fun showSearchResult(result: List<SearchResult>) {
+        searchAdapter.setItems(result)
+        showEmptyState(isEmpty = result.isEmpty())
+    }
+
+    override fun showMessage(textRes: Int?) = with(binding.messageTextView) {
+        isVisible = if (textRes == null) {
+            false
+        } else {
+            setText(textRes)
+            true
+        }
```

Comment:
`binding.messageTextView withTextOrGone textRes`

Reply:
ohh you right! thx!